Automattic/wp-calypso
# Store verticals list per segment instead of one global list

#### Changes proposed in this Pull Request

Since we're adding a support for Professional and Blogger site types, we need a way to store segment-specific verticals. This PR updates all relevant places and changes the state structure from:

```js
signup: {
    verticals: {
        restaurant: [
             { /* ... */ }
        ]
    }
}
```

To

```js
signup: {
    verticals: {
        4: {
           restaurant: [
             { /* ... */ }
           ]
        },
        1: {
           restaurant: [
             { /* ... */ }
           ]
        }
    }
}
```

#### Testing instructions

This is just a state layer update so there isn't a real new feature to test. Instead we're just confirming nothing got broken in the process:

1. Go to `/start/onboarding` and choose a business segment.
2. Type in and select a few different verticals, confirm in Network tab that http request contains `site_type=1` (or any other segment ID) in query
3. Use a back button and choose business segment again, confirm the site preview still works
4. Inspect state and the following actions in redux devtools: `SIGNUP_VERTICALS_SET`, `SIGNUP_VERTICALS_REQUEST`. Confirm they contain `siteTypeId` when verticals lookup is performed. Confirm the state is shaped like in the PR description above.
5. Finish the signup, confirm a proper starter site was created.
6. Go to `/start/main` and confirm verticals picker works there. Finish she signup flow, confirm a new site was created.
